bool state # true=attached collision object is on
---
bool completed
string msg